java-api/src/main/c/generated/tiledb_wrap.cxx

```diff
@@ -1959,6 +1959,138 @@ extern "C" {
     return sizeof(uint64_t);
   }
 
+  struct Callback {
+     jobject obj;
+     jclass cls;
+  };
+
+  SWIGEXPORT JavaVM* getJVM(){
+    JavaVM* vm;
+    jsize vmCount;
+    if (JNI_GetCreatedJavaVMs(&vm, 1, &vmCount) != JNI_OK || vmCount == 0) {
+        fprintf(stderr, "Could not get active VM\n");
+        return NULL;
+    }
+    return vm;
+  }
+
+  SWIGEXPORT JNIEnv* getJNI(JavaVM* vm){
+    JNIEnv* env;
+    jint result = vm->GetEnv((void**)&env, JNI_VERSION_1_6);
+    if (result == JNI_EDETACHED) {
+        result = vm->AttachCurrentThread((void**)&env, NULL);
+    }
+    if (result != JNI_OK) {
+        fprintf(stderr, "Failed to get JNIEnv\n");
+        return NULL;
+    }
+    return env;
+  }
+
+  SWIGEXPORT void java_callback(void* data) {
+    JavaVM* vm = getJVM();
+    JNIEnv* jenv = getJNI(vm);
+    jobject obj = ((Callback*)data)->obj;
+    jclass cls=jenv->GetObjectClass(obj);
+    jmethodID mid = jenv->GetMethodID(cls, "call", "()V");
+    if (mid == 0)
+      return;
+    jenv->CallVoidMethod(obj, mid);
+    vm->DetachCurrentThread();
+  }
+
+  SWIGEXPORT int java_path_callback(const char* path, tiledb_object_t type, void* data) {
+    JavaVM* vm = getJVM();
+    JNIEnv* jenv = getJNI(vm);
+    jobject obj = ((Callback*)data)->obj;
+    jclass cls=jenv->GetObjectClass(obj);
+    jmethodID mid = jenv->GetMethodID(cls, "call", "(JI)I");
+    if (mid == 0)
+      return -1;
+    jlong jpath;
+    *(const char **)&jpath = path;
+    jint ret = jenv->CallIntMethod(obj, mid, jpath, (jint) type);
+    vm->DetachCurrentThread();
+    return ret;
+  }
+
+  JNIEXPORT jint JNICALL Java_io_tiledb_libtiledb_UtilsJNI_tiledb_1query_1submit_1async(JNIEnv *jenv, jclass jcls, jlong jarg1, jobject jarg1_, jlong jarg2, jobject jarg2_, jobject jarg3) {
```

Comment:
Instead of these being in a separate class, could we add them to `tiledbJNI` so that there is only one native binding class to deal with?